src/daemon/mod.rs

```diff
@@ -278,8 +277,6 @@ pub(super) async fn start(
         return Ok(());
     }
 
-    let epoch = chain_store.heaviest_tipset().epoch();
-
```

Comment:
Why did you remove this?

Reply:
Because logically it's insufficient to use `epoch` to make the decision in `fetch_snapshot_if_required`, ideally it should validate the integrity of the union DB (the change is not included in this PR as suggested)

Reply:
I don't understand why it's logically insufficient. If it _is_ insufficient, document why in the code. If it _isn't_, let's revert the code.

Reply:
Code doc added [here](https://github.com/ChainSafe/forest/pull/3365/files#diff-bf435d9dfa9a63896065cbbd3082d65b24a937afbab2638ae820385108446a59R479)
```rust
// TODO: Add integrity check of the union DB to decide whether or not a snapshot is required.
// Checking the heaviest tipset that is only loaded from parity-db is insufficient
```

Reply:
You only wrote that it is insufficient, but you haven't written why it is insufficient. I do not believe it is insufficient. Write a comment that convinces me otherwise.

Reply:
@lemmih  The first part of the comment is the reason. We require a snapshot when forest cannot run with the current union DB, successfully reading epoch info from parity-db does not mean DB is complete (e.g. CAR DB files are manually removed). Does that make sense?

Reply:
Reverted as suggested